src/modules/fw_path_navigation/FixedwingPositionControl.cpp

```diff
@@ -1068,7 +1036,7 @@ FixedwingPositionControl::control_auto_position(const float control_interval, co
 
 	} else {
 		// No valid previous waypoint, go for the current wp.
-		// This is automatically handled by the L1/NPFG libraries.
+		// This is automatically handled by the NPFG libraries.
```

Comment:
I would remove the hacky superpositioned waypoints here and rather either call navigateWaypoints() (when there is a valid previous point) or navigateBearing() when there is only the one point (calculating the bearing from aircraft current position to singular waypoint).

Reply:
I fix it, but instead of navigateBearing i do navigateWaypoints where the previous waypoint is the aircraft position. Should be very similar to the navigateBearing function if the dt is small and would at least make sure to hit the current waypoint. Similar to the changes made in https://github.com/PX4/PX4-Autopilot/pull/21071